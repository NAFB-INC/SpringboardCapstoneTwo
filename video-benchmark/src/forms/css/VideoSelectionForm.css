.VideoSelectionForm{
    position:absolute;
    margin-top:5%;
    margin-left:5%;
    margin-right:5%;
    padding-top:2%;
    height:70%;
    width:88%;
    border:5px solid #6D2E46;
    background-color: #ECE2D0;
}

.red{
    color:red;
}