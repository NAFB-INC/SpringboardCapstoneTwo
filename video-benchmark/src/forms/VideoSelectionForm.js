import "./css/VideoSelectionForm.css";
import React, { useState } from "react";
import useFields from "../hooks/useFields";

function VideoSelectionForm() {
    const [errorsCode,setErrorsCode] = useState([]);
    const [errorsSearch,setErrorsSearch] = useState([]);
    const [formData, handleChange, resetForm] = useFields({
        audience_code: '',
        key_phrase: ''
    })

    const handleSubmitCode = e => {
        e.preventDefault();
        resetForm();
    }
    
    const handleSubmitSearch = e => {
        e.preventDefault();
        resetForm();
    }

    return (
        <div className="VideoSelectionForm">
            <div className="form-text">Have a presentation code? Find what you are looking for right away with that!</div>
            <form onSubmit={handleSubmitCode}>
                <div className="form-box">
                    <label htmlFor="audience_code">Code: </label>
                    <input name="audience_code" type="text" placeholder="abc123" value={formData.audience_code} onChange={handleChange}></input>
                </div>
                <div>
                    {errorsCode.map((e)=>(
                        <li className="error" key={`error${e}`}>{e}</li>
                    ))}
                </div>
                <button className="submit-btn">Submit</button>
            </form>
            <br/>
            <hr/>
            <br/>
            <div className="form-text">OR maybe you are looking for a person or topic?</div>
            <form onSubmit={handleSubmitSearch}>
                <div className="form-box">
                    <label htmlFor="key_phrase">Phrase: </label>
                    <input name="key_phrase" type="text" placeholder="presentations" value={formData.key_phrase} onChange={handleChange}></input>
                </div>
                <div>
                    {errorsSearch.map((e)=>(
                        <li className="error" key={`error${e}`}>{e}</li>
                    ))}
                </div>
                <button className="submit-btn">Submit</button>
            </form>
        </div>
    );
}

export default VideoSelectionForm;