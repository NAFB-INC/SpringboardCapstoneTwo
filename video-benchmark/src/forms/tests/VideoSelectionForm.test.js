import React from "react";
import { render } from '@testing-library/react';
import VideoSelectionForm from '../VideoSelectionForm';

it("renders without crashing", function() {
  render(<VideoSelectionForm />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<VideoSelectionForm />);
  expect(asFragment()).toMatchSnapshot();
});