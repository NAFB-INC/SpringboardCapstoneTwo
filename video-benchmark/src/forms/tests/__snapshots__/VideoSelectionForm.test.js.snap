// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="VideoSelectionForm"
  >
    <div
      class="form-text"
    >
      Have a presentation code? Find what you are looking for right away with that!
    </div>
    <form>
      <div
        class="form-box"
      >
        <label
          for="audience_code"
        >
          Code: 
        </label>
        <input
          name="audience_code"
          placeholder="abc123"
          type="text"
          value=""
        />
      </div>
      <div />
      <button
        class="submit-btn"
      >
        Submit
      </button>
    </form>
    <br />
    <hr />
    <br />
    <div
      class="form-text"
    >
      OR maybe you are looking for a person or topic?
    </div>
    <form>
      <div
        class="form-box"
      >
        <label
          for="key_phrase"
        >
          Phrase: 
        </label>
        <input
          name="key_phrase"
          placeholder="presentations"
          type="text"
          value=""
        />
      </div>
      <div />
      <button
        class="submit-btn"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
